server/sonar-server/src/main/java/org/sonar/server/user/ws/SearchAction.java

```diff
@@ -67,18 +81,34 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn().checkGlobalPermission(GlobalPermissions.SYSTEM_ADMIN);
+
     SearchOptions options = new SearchOptions()
       .setPage(request.mandatoryParamAsInt(Param.PAGE), request.mandatoryParamAsInt(Param.PAGE_SIZE));
     List<String> fields = request.paramAsStrings(Param.FIELDS);
     SearchResult<UserDoc> result = userIndex.search(request.param(Param.TEXT_QUERY), options);
 
+    Multimap<String, String> groupsByLogin = Multimaps.forMap(Maps.<String, String>newHashMap());
```

Comment:
I don't care about the returned implementation. I could have used e.g ArrayListMultimap.create().